Kernel/Arch/aarch64/RPi/InterruptController.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022, Timon Kruiper <timonkruiper@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Format.h>
+#include <Kernel/Arch/aarch64/RPi/InterruptController.h>
+#include <Kernel/Arch/aarch64/RPi/MMIO.h>
+#include <Kernel/Interrupts/GenericInterruptHandler.h>
+
+namespace Kernel::RPi {
+
+// "7.5 Interrupts Registers"
+// https://github.com/raspberrypi/documentation/files/1888662/BCM2837-ARM-Peripherals.-.Revised.-.V2-1.pdf
+struct InterruptControllerRegisters {
+    u32 IRQ_basic_pending;
+    u32 IRQ_pending_1;
+    u32 IRQ_pending_2;
+    u32 FIQ_control;
+
+    u32 enable_IRQs_1;
+    u32 enable_IRQs_2;
+    u32 enable_basic_IRQs;
+
+    u32 disable_IRQs_1;
+    u32 disable_IRQs_2;
+    u32 disable_basic_IRQs;
```

Comment:
We rarely use uppercase in identifiers, and I don't see why it would be needed here - `foo_irqs` seems just as clear.

Reply:
I got these names from the documentation, but smaller case is better indeed!